test-data/unit/check-classes.test

```diff
@@ -2617,6 +2617,27 @@ b = a.bar
 [out]
 main:9: error: Incompatible types in assignment (expression has type "A", variable has type "B")
 
+[case testDecoratedGetAttribute]
+from typing import Callable, TypeVar
+
+T = TypeVar('T', bound=Callable)
+
+def decorator(f: T) -> T:
+    return f
```

Comment:
Test a decorator that changes the signature of the method. I this might not work correctly right now (for example, return plain `Callable` -> it should turn the attribute type to `Any`. If this is an issue, you'll probably need to get the method type from the decorator node.

Reply:
It is working correctly, no need to change that. Thanks for the idea 👍

Reply:
Nope, I was wrong. I've misinterpreted your words in an opposite way.

Example of test that passes, but should not: https://github.com/python/mypy/pull/11150/commits/28d950d10b39e9c09865e7d4f62ab437152ba238#diff-1c26dc582ed5f36723414080dfb646bc71cf20736b73ae1ccff9b90a5d3befafR2635-R2638

Right now this happens:

```python
from typing import Callable, TypeVar

T = TypeVar('T', bound=Callable)

def decorator(f: Callable[['A', str], 'A']) -> Callable[['A', str], int]:
    def inner(self: 'A', x: str) -> int:
        return f(self, x).num
    return inner

class A:
    num = 1

    @decorator
    def __getattr__(self, x: str) -> 'A':
        return A()

a = A()

reveal_type(a.a)  # Revealed type is "ex.A"
# In runtime: 1
```

So, yeah. I need to get type from `Decorator`, not just its node.